cmd/koslet/app/koslet.go

```diff
@@ -0,0 +1,113 @@
+package app
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/kosmos.io/kosmos/cmd/koslet/app/options"
+	"github.com/kosmos.io/kosmos/pkg/koslet"
+)
+
+const (
+	specNodeNameField = "spec.nodeName"
+)
+
+// NewKosletCommand creates a *cobra.Command object with default parameters
+func NewKosletCommand(ctx context.Context) *cobra.Command {
+	opts := options.NewOptions()
+
+	cmd := &cobra.Command{
+		Use:  "koslet",
+		Long: `The koslet is running in each node and responsible for QoS ensurance`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// validate options
+			if errs := opts.Validate(); len(errs) != 0 {
+				return errs.ToAggregate()
+			}
+			if err := Run(ctx, opts); err != nil {
+				return err
+			}
+			return nil
+		},
+		Args: func(cmd *cobra.Command, args []string) error {
+			for _, arg := range args {
+				if len(arg) > 0 {
+					return fmt.Errorf("%q does not take any arguments, got %q", cmd.CommandPath(), args)
+				}
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().AddGoFlagSet(flag.CommandLine)
+	opts.AddFlags(cmd.Flags())
+	utilfeature.DefaultMutableFeatureGate.AddFlag(cmd.Flags())
+
+	return cmd
+}
+
+func Run(ctx context.Context, opts *options.Options) error {
+	nodename := getHostName(opts.NodeName)
+	kubeClient, err := buildClient()
+	if err != nil {
+		return err
+	}
+
+	informerFactory := informers.NewSharedInformerFactory(kubeClient, time.Second*30)
+	pvInformer := informerFactory.Core().V1().PersistentVolumes()
+	pvcInformer := informerFactory.Core().V1().PersistentVolumeClaims()
+
+	podInformerFactory := informers.NewSharedInformerFactoryWithOptions(kubeClient, time.Minute,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.FieldSelector = fields.OneTermEqualSelector(specNodeNameField, nodename).String()
+		}),
+	)
+	podInformer := podInformerFactory.Core().V1().Pods()
+	newKoslet, err := koslet.NewKoslet(ctx, nodename, kubeClient, pvInformer, pvcInformer, podInformer, opts.KubeletRootPath, opts.DiskIOInitConfig)
+
+	if err != nil {
+		return err
+	}
+
+	podInformerFactory.Start(ctx.Done())
+	podInformerFactory.WaitForCacheSync(ctx.Done())
+	newKoslet.Run()
+	return nil
+}
+
+func buildClient() (kubernetes.Interface, error) {
```

Comment:
Can we unify the clientset building process or we can put it into a common utils

Reply:
Okay, I'll modify it